api/api_ethereum_test.go

```diff
@@ -0,0 +1,158 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"reflect"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+
+	mock_api "github.com/klaytn/klaytn/api/mocks"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"github.com/klaytn/klaytn/consensus/mocks"
+	"github.com/klaytn/klaytn/crypto"
+	"github.com/klaytn/klaytn/networks/rpc"
+	"github.com/klaytn/klaytn/params"
+	"github.com/klaytn/klaytn/rlp"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestEthereumAPI_GetHeaderByNumber tests GetHeaderByNumber.
+func TestEthereumAPI_GetHeaderByNumber(t *testing.T) {
+	testGetHeader(t, "GetHeaderByNumber")
+}
+
+// TestEthereumAPI_GetHeaderByHash tests GetHeaderByNumber.
+func TestEthereumAPI_GetHeaderByHash(t *testing.T) {
+	testGetHeader(t, "GetHeaderByHash")
+}
+
+// testGetHeader generates data to test GetHeader related functions in EthereumAPI
+// and actually tests the API function passed as a parameter.
+func testGetHeader(t *testing.T, testAPIName string) {
+	mockCtrl := gomock.NewController(t)
+	mockBackend := mock_api.NewMockBackend(mockCtrl)
+
+	api := EthereumAPI{
+		publicKlayAPI:       NewPublicKlayAPI(mockBackend),
+		publicBlockChainAPI: NewPublicBlockChainAPI(mockBackend),
+	}
+
+	// Creates a MockEngine.
+	mockEngine := mocks.NewMockEngine(mockCtrl)
+	// GetHeader APIs calls internally below methods.
+	mockBackend.EXPECT().Engine().Return(mockEngine)
+	// Author is called when calculates miner field of Header.
+	mockEngine.EXPECT().Author(gomock.Any()).Return(common.Address{}, nil)
+	mockBackend.EXPECT().GetTd(gomock.Any()).Return(new(big.Int))
+
+	header := types.CopyHeader(&types.Header{}) // Creates an empty header
```

Comment:
Why don't you put values in this header? Without this, the following tests only check field existence and dummy values after calling `rpcMarshalHeader`. If this header contains proper values corresponding to the json example below, this TC can test whether `rpcMarshalHeader` assignes all field values properly.

Reply:
@aidan-kwon 
Thanks for your feedback.
As you said, I should have been written test cases like you describe.
Applied your feedback [here](https://github.com/klaytn/klaytn/pull/1088/commits/9db424abe98f41656def6bc7693e93068ef6b4be).

Now TCs can checks whether `rpcMarshalHeader` assigns all field values properly or not.
In the [line 157 - 177](https://github.com/klaytn/klaytn/pull/1088/commits/9db424abe98f41656def6bc7693e93068ef6b4be#diff-5929e238aaa795539ea5f0eb411cd553e712ae2bc234cdf6cde9f28257c87149R157-R177) we can make sure that `rpcMarshalHeader` didn't modify it's original values of header.

Reply:
@aidan-kwon 
Refactored test case with just checking json marshaled data [here](https://github.com/klaytn/klaytn/pull/1088/commits/521fb3c022a7f020bbff4cc370c8685acfd7a2ad).

Thanks.